src/infra/src/table/migration/m20250109_085148_alter_dashboards_folder_id.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 OpenObserve Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Drops the folder's created_at column since Sea ORM maps this column's data
+//! type to different Rust types creating runtime errors when running on a
+//! non-PostgreSQL database.
+
+use sea_orm::{
+    ActiveModelTrait, ColumnTrait, EntityTrait, PaginatorTrait, QueryFilter, Set, TransactionTrait,
+};
+use sea_orm_migration::prelude::*;
+
+const DASHBOARDS_FOLDERS_FK: &str = "dashboards_folders_fk";
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        add_nullable_folder_uuid_column(manager).await?;
+        update_folder_uuid_column(manager).await?;
+        drop_foreign_key(manager).await?;
+        delete_folder_id_column(manager).await?;
+        rename_folder_uuid_column(manager).await?;
+        Ok(())
+    }
+
+    async fn down(&self, _manager: &SchemaManager) -> Result<(), DbErr> {
+        // Reversing this migration is not supported.
+        Ok(())
+    }
+}
+
+// Removes the old created_at column.
+async fn add_nullable_folder_uuid_column(manager: &SchemaManager<'_>) -> Result<(), DbErr> {
+    manager
+        .alter_table(
+            Table::alter()
+                .table(Dashboards::Table)
+                .add_column(
+                    ColumnDef::new(Dashboards::FolderUuid)
+                        .string_len(256)
+                        .default(Expr::value(""))
+                        .not_null(),
+                )
+                .to_owned(),
+        )
+        .await?;
+    Ok(())
+}
+
+// Removes the old created_at column.
+async fn update_folder_uuid_column(manager: &SchemaManager<'_>) -> Result<(), DbErr> {
+    let db = manager.get_connection();
+    let txn = db.begin().await?;
+    let mut dashboard_pages = dashboards::Entity::find().paginate(&txn, 100);
+    while let Some(dashboards) = dashboard_pages.fetch_and_next().await? {
+        for dashboard in dashboards.into_iter() {
+            let folder = folders::Entity::find()
+                .filter(folders::Column::Id.eq(dashboard.folder_id))
+                .one(db)
+                .await?
+                .unwrap();
```

Comment:
Nit:

Rather than fetching each individual `folder` for each dashboard, I believe you can replace

```
dashboards::Entity::find().paginate(&txn, 100);
```

with something  like 

```
dashboards::Entity::find().find_also_related(folders::Entity).paginate(&txn, 100);
```
(docs [here](https://www.sea-ql.org/SeaORM/docs/basic-crud/select/#one-to-one))

with `find_also_related` SeaORM will generate a `join` query so it wouldn't be necessary to make a separate request to the DB to fetch each related folder individually

Reply:
All that being said, given that this is just a migration, it might not really matter if this migration ends up hitting the DB more than necessary